### PR TITLE
NFDeployment Kpt function changes 

### PR DESCRIPTION
**Pull request by aakashchan**
#402 
1. Updated the Nf Deploy Fn to use latest API changes by removing older generics code.
2. Removed reference to all the older APIs in testdata including other functions
3. Added tests for dependency test cases. 

Tests Run
1.   Executed all pipeline tests - *Pass*
2.  Executed dependency tests for single and multiple - *Pass*
3.  Execute all functions unite tests - *Pass*
4.  Docker build  - *Pass*

Docker image for testing - `docker.io/lostbrain/nfdeployfn:test-1`

### CODE REVIEW

**Comment by aakashchan** [comment]
Notes: 

1. I am not able to replicate the multiple duplicate config entries bug  @arora-sagar So didn't want to add duplicate removal code in kpt function without finding the root cause.

2. Not able to reproduce the same name for multiple dependencies. Have added that tests too, you can verify the expected data. @josephthaliath 

3. Assumption is provider field will be present in base package and also have not added validation for the same. @henderiw

**Comment by henderiw** [comment]
update the go mod files for all the fns to include the latest api. This will solve some test failures

**Comment by henderiw** [comment]
The one thing I see and this is a discussion. Since we allow for parameterRefs to be present from the beginning. Should we do the same for interfaces, dnns, etc ?

This allows the package builder to add things statically w/o the need for specialization ?

**Comment by josephthaliath** [comment]
Thanks @aakashchan for checking the multiple dependencies. I think it was an issue on our side. Just wanted to check if we can add ParametersRef statically in the package, will krm function overwrite this while generating parameterRefs for dependencies added.

File: krm-functions/nfdeploy-fn/README.md [hunk 1]
```diff
@@ -1,18 +1,18 @@
...
 The upf deployment function has 3 `watch` resources:
 - `req.nephio.org/v1alpha1.Interface` 
-   - For every interface resource status, it will use `status.ipAllocationStatus` and `status.vlanAllocationStatus` fields to populate UPFDeployment `spec.interfaces` values.
+   - For every interface resource status, it will use `status.ipAllocationStatus` and `status.vlanAllocationStatus` fields to populate NFDeployment `spec.interfaces` values.
 - `req.nephio.org/v1alpha1.DataNetwork`
-    - For every datanetwork resource status, it will use `status.pools.ipAllocation` fields to populate UPFDeployment `spec.networkInstances` values.
+    - For every datanetwork resource status, it will use `status.pools.ipAllocation` fields to populate NFDeployment `spec.networkInstances` values.
 - `req.nephio.org/v1alpha1.Capacity`
-    - This resource is the source of truth for the capacity details in UPFDeployment spec.
+    - This resource is the source of truth for the capacity details in NFDeployment spec.
```
Comment (henderiw) [inline krm-functions/nfdeploy-fn/README.md:13]: add the DependencyKind as well here

**Review by henderiw — CHANGES_REQUESTED** [review]

**Comment by gvbalaji** [comment]
Looks like this PR is approved but needs a rebase @aakashchan

**Comment by aakashchan** [comment]
> Looks like this PR is approved but needs a rebase @aakashchan

Yes, done. We are good to merge now. Thanks @gvbalaji

**Pull request merged**